crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -428,12 +428,21 @@ where
 {
     choice((
         constrain(expr_parser.clone()),
+        assertion(expr_parser.clone()),
         declaration(expr_parser.clone()),
         assignment(expr_parser.clone()),
         expr_parser.map(Statement::Expression),
     ))
 }
 
+fn assertion<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
```

Comment:
Another way we could have written it is by doing something like:

```rust
fn assertion<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
where
    P: ExprParser + 'a,
{
    ignore_then_commit(
        choice((keyword(Keyword::Assert), keyword(Keyword::Constrain))).labelled("statement"),
        expr_parser,
    )
    .map(|expr| Statement::Constrain(ConstrainStatement(expr)))
}
```

Where `choice` essentially means "or".

Then we can remove the `constrain`  function and its tests. We then would not have duplicated code between the assert and constrain test case.